
#include <fstream>
#include "TMV_Test.h"
#include "TMV_Test_1.h"

bool XXDEBUG1 = false;
bool XXDEBUG2 = false;
bool XXDEBUG3 = false;
bool XXDEBUG4 = false;
bool XXDEBUG5 = false;
bool XXDEBUG6 = false;
bool XXDEBUG7 = false;
bool XXDEBUG8 = false;
bool XXDEBUG9 = false;

bool showtests = false;
bool showacc = false;
bool showdiv = false;
bool showstartdone = false;
bool donorm2 = true;
bool symoprod = false;
bool dontthrow = false;
std::string lastsuccess = "";

int main() try 
{
    std::ofstream log("tmvtest1d.log");
    tmv::WriteWarningsTo(&log);

    //showacc=true;
    //showdiv=true;
    //showtests=true;
    //showstartdone=true;
 
#if 1

#ifdef TEST_DOUBLE
    TestMatrixDiv<double>();
    TestMatrixDet<double>();
#endif // DOUBLE

#ifdef TEST_FLOAT
    TestMatrixDiv<float>();
    TestMatrixDet<float>();
#endif // FLOAT

#ifdef TEST_INT
    TestMatrixDet<int>();
#endif // INT

#ifdef TEST_LONGDOUBLE
    TestMatrixDiv<long double>();
    TestMatrixDet<long double>();
#endif // LONGDOUBLE

#endif

    return 0;
}
#if 1
#ifndef NOTHROW
catch (tmv::Error& e) {
    std::cerr<<e<<std::endl;
    std::cerr<<"Last successful test was "<<lastsuccess<<std::endl;
    return 1;
}
#endif
catch (std::exception& e) {
    std::cerr<<e.what()<<std::endl;
    std::cerr<<"Last successful test was "<<lastsuccess<<std::endl;
    return 1;
}
catch (...) {
    std::cerr<<"Unknown exception thrown\n";
    std::cerr<<"Last successful test was "<<lastsuccess<<std::endl;
    return 1;
}
#else
catch (int) {}
#endif

